shortcuts.js

```diff
@@ -0,0 +1,391 @@
+const AppConstants = require("resource://gre/modules/AppConstants.jsm");
+const { Cc, Ci, Cu } = require("chrome");
+const {Services} = require("resource://gre/modules/Services.jsm");
+
+const NEW_TAB_TIMEOUT = 300;
+
+function getBrowserURL() {
+  return "chrome://browser/content/browser.xul";
+}
+
+function whereToOpenLink(e, ignoreButton, ignoreAlt) {
+  // This method must treat a null event like a left click without modifier keys (i.e.
+  // e = { shiftKey:false, ctrlKey:false, metaKey:false, altKey:false, button:0 })
+  // for compatibility purposes.
+  if (!e)
+    return "current";
+
+  var shift = e.shiftKey;
+  var ctrl =  e.ctrlKey;
+  var meta =  e.metaKey;
+  var alt  =  e.altKey && !ignoreAlt;
+
+  // ignoreButton allows "middle-click paste" to use function without always opening in a new window.
+  var middle = !ignoreButton && e.button == 1;
+  var middleUsesTabs = Services.prefs.getBoolPref("browser.tabs.opentabfor.middleclick", true);
+
+  // Don't do anything special with right-mouse clicks.  They're probably clicks on context menu items.
+
+  var metaKey = AppConstants.platform == "macosx" ? meta : ctrl;
+  if (metaKey || (middle && middleUsesTabs))
+    return shift ? "tabshifted" : "tab";
+
+  if (alt && Services.prefs.getBoolPref("browser.altClickSave", false))
+    return "save";
+
+  if (shift || (middle && !middleUsesTabs))
+    return "window";
+
+  return "current";
+}
+
+function BrowserOpenTab(event, win) {
+  let where = "tab";
+  let relatedToCurrent = false;
+  //let doc = event.target.ownerDocument;
+  //let win = doc.defaultView;
+
+  if (event) {
+    where = whereToOpenLink(event, false, true);
+
+    switch (where) {
+      case "tab":
+      case "tabshifted":
+        // When accel-click or middle-click are used, open the new tab as
+        // related to the current tab.
+        relatedToCurrent = true;
+        break;
+      case "current":
+        where = "tab";
+        break;
+    }
+  }
+
+  openUILinkIn(win.BROWSER_NEW_TAB_URL, where, { relatedToCurrent }, undefined, undefined, win);
+}
+function openUILinkIn(url, where, aAllowThirdPartyFixup, aPostData, aReferrerURI, win) {
```

Comment:
we don't use the aVar in this file (and in general in this addon). Rename aAllowThirdPartyFixup, aPostData, aReferrerURI to allowThirdPartyFixup, postData and referrerURI